chore: use TypeScript v5 to build

Usage of TS v5 generates tsc errors: see #185 and #186

```
Version 5.0.2
Error: src/view/mixins/ValidationMixin.ts(324,10): error TS2365: Operator '>' cannot be applied to types 'number' and 'string | number'.
Error: src/view/mixins/ValidationMixin.ts(330,10): error TS2365: Operator '>' cannot be applied to types 'number' and 'string | number'.
Error: src/view/other/Multiplicity.ts(160,45): error TS2365: Operator '>=' cannot be applied to types 'number' and 'string | number'.
Error: src/view/other/Multiplicity.ts(161,47): error TS2365: Operator '>=' cannot be applied to types 'number' and 'string | number'.
Error: Process completed with exit code 2.
```

To review as part of this issue
- the `html` package declares a typescript dependency whereas it is not using TypeScript
- the `core` package declares a typescript dependency to workaround a TS error because TS v5 was resolved instead of v4 (#186)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,10 @@
 # `maxGraph` Change Log
 
+## UNRELEASED
+
+**Breaking Changes**
+- it is no longer possible to pass a 'n' value for the `max` property of the `Multiplicity` class. Pass `null` instead to have the same effect.
+
 ## 0.8.0
 
 Release date: `2024-02-14`

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "eslint-plugin-import": "~2.29.1",
     "eslint-plugin-prettier": "~5.1.3",
     "prettier": "~3.1.1",
-    "typescript": "^4.9.5"
+    "typescript": "~5.3.2"
   },
   "overrides": {
     "@types/node": "^16.18.0"

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "eslint-plugin-import": "~2.29.1",
     "eslint-plugin-prettier": "~5.1.3",
     "prettier": "~3.1.1",
-    "typescript": "~5.3.2"
+    "typescript": "~5.3.3"
   },
   "overrides": {
     "@types/node": "^16.18.0"

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -51,7 +51,7 @@
     "npm-run-all": "~4.1.5",
     "rimraf": "~5.0.5",
     "ts-jest": "^29.0.3",
-    "typedoc": "~0.25.4"
+    "typedoc": "~0.25.8"
   },
   "sideEffects": true
 }

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -51,8 +51,7 @@
     "npm-run-all": "~4.1.5",
     "rimraf": "~5.0.5",
     "ts-jest": "^29.0.3",
-    "typedoc": "^0.23.21",
-    "typescript": "^4.9.5"
+    "typedoc": "~0.25.4"
   },
   "sideEffects": true
 }

--- a/packages/core/src/view/Graph.ts
+++ b/packages/core/src/view/Graph.ts
@@ -113,8 +113,7 @@ class Graph extends EventSource {
   mouseListeners: MouseListenerSet[] = [];
 
   /**
-   * An array of {@link Multiplicity} describing the allowed
-   * connections in a graph.
+   * An array of {@link Multiplicity} describing the allowed connections in a graph.
    */
   multiplicities: Multiplicity[] = [];
 

--- a/packages/core/src/view/other/Multiplicity.ts
+++ b/packages/core/src/view/other/Multiplicity.ts
@@ -47,7 +47,7 @@ class Multiplicity {
     attr: string,
     value: string,
     min: number | null | undefined,
-    max: number | 'n' | null | undefined,
+    max: number | null | undefined,
     validNeighbors: string[],
     countError: string,
     typeError: string,
@@ -57,8 +57,8 @@ class Multiplicity {
     this.type = type;
     this.attr = attr;
     this.value = value;
-    this.min = min != null ? min : 0;
-    this.max = max != null ? max : 'n';
+    this.min = min ?? 0;
+    this.max = max ?? Number.MAX_VALUE;
     this.validNeighbors = validNeighbors;
     this.countError = Translations.get(countError) || countError;
     this.typeError = Translations.get(typeError) || typeError;
@@ -92,17 +92,15 @@ class Multiplicity {
 
   /**
    * Defines the minimum number of connections for which this rule applies.
-   *
    * @default 0
    */
   min: number;
 
   /**
    * Defines the maximum number of connections for which this rule applies.
-   * A value of 'n' means unlimited times.
-   * @default 'n'
+   * @default {@link Number.MAX_VALUE}
    */
-  max: number | 'n';
+  max: number;
 
   /**
    * Holds an array of strings that specify the type of neighbor for which

--- a/packages/html/package.json
+++ b/packages/html/package.json
@@ -10,13 +10,13 @@
     "preview": "vite preview"
   },
   "devDependencies": {
-    "@storybook/addon-essentials": "^7.4.6",
-    "@storybook/addon-links": "^7.4.6",
-    "@storybook/addon-storysource": "^7.4.6",
-    "@storybook/blocks": "^7.4.6",
-    "@storybook/html": "^7.4.6",
-    "@storybook/html-vite": "^7.4.6",
-    "storybook": "^7.4.6",
+    "@storybook/addon-essentials": "~7.6.17",
+    "@storybook/addon-links": "~7.6.17",
+    "@storybook/addon-storysource": "~7.6.17",
+    "@storybook/blocks": "~7.6.17",
+    "@storybook/html": "~7.6.17",
+    "@storybook/html-vite": "~7.6.17",
+    "storybook": "~7.6.17",
     "vite": "^5.0.12"
   }
 }

--- a/packages/html/package.json
+++ b/packages/html/package.json
@@ -17,7 +17,6 @@
     "@storybook/html": "^7.4.6",
     "@storybook/html-vite": "^7.4.6",
     "storybook": "^7.4.6",
-    "typescript": "^5.2.2",
     "vite": "^5.0.12"
   }
 }

--- a/packages/website/package.json
+++ b/packages/website/package.json
@@ -26,8 +26,7 @@
   "devDependencies": {
     "@docusaurus/module-type-aliases": "3.0.0",
     "@docusaurus/tsconfig": "3.0.0",
-    "@docusaurus/types": "3.0.0",
-    "typescript": "~5.2.2"
+    "@docusaurus/types": "3.0.0"
   },
   "browserslist": {
     "production": [

